trie/trieStorageManager.go

```diff
@@ -133,29 +139,61 @@ func NewTrieStorageManager(args NewTrieStorageManagerArgs) (*trieStorageManager,
 	tsm.snapshots = snapshots
 	tsm.snapshotId = snapshotId
 
-	go tsm.doCheckpointsAndSnapshots(ctx, args.Marshalizer, args.Hasher)
+	go tsm.doCheckpointsAndSnapshots(ctx, args.Marshalizer, args.Hasher, goRoutinesThrottler)
 	return tsm, nil
 }
 
-func (tsm *trieStorageManager) doCheckpointsAndSnapshots(ctx context.Context, msh marshal.Marshalizer, hsh hashing.Hasher) {
-	tsm.doProcessLoop(ctx, msh, hsh)
+func (tsm *trieStorageManager) doCheckpointsAndSnapshots(ctx context.Context, msh marshal.Marshalizer, hsh hashing.Hasher, goRoutinesThrottler core.Throttler) {
+	tsm.doProcessLoop(ctx, msh, hsh, goRoutinesThrottler)
 	tsm.cleanupChans()
 }
 
-func (tsm *trieStorageManager) doProcessLoop(ctx context.Context, msh marshal.Marshalizer, hsh hashing.Hasher) {
+func (tsm *trieStorageManager) doProcessLoop(ctx context.Context, msh marshal.Marshalizer, hsh hashing.Hasher, goRoutinesThrottler core.Throttler) {
+	defer log.Debug("trieStorageManager.storageProcessLoop go routine is closing...")
+
 	for {
 		select {
 		case snapshotRequest := <-tsm.snapshotReq:
-			tsm.takeSnapshot(snapshotRequest, msh, hsh, ctx)
+			err := checkGoRoutinesThrottler(ctx, goRoutinesThrottler)
+			if err != nil {
+				tsm.finishOperation(snapshotRequest, "did not start snapshot, goroutione is closing")
```

Comment:
this is a bit odd. You assume that if the checkGoRoutinesThrottler errors then the go routine is closing. 
Strongly suggest to:
1. attach the checkGoRoutinesThrottler method to the trieStorageManager
2. call the finishOperation operation with this message on the L190 branch

Reply:
Done.